extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -628,10 +504,143 @@ public void onValueReceived(String variable, String value, String service) {
 
     }
 
-    @Override
-    public void channelLinked(ChannelUID channelUID) {
-        if (channelUID.getId().equalsIgnoreCase(COORDINATOR)) {
-            updateGroupCoordinator();
+    protected void updateChannel(String channeldD) {
+        State newState = UnDefType.UNDEF;
+        switch (channeldD) {
+            case STATE:
+                updateState(STATE, (stateMap.get("TransportState") != null)
```

Comment:
You are using different logic in the case statements. Wouldn't it be more consistent to use one for all?

newState is already set to undef.

Here for `STATE` you don't use newState, but use the trinary operator in the update state method.

Below for `CONTROL` you first look if the state info is not null and then you don't override newState but set the state in the resp. sections. So no undef is set at all.

Below for `SHUFFLE` you set the newState it the state info is not null and then (in both cases) you set the newState (so undef or the known one).